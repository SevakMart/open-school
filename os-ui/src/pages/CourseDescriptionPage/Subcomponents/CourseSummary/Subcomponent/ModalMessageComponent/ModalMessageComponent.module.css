.mainContainer {
  width: 100%;
  height: 100%;
  background-image: conic-gradient(at 50% 38%, #EEEFF4 0deg, #EEEFF4 60deg, #FFFFFF 60deg, #FFFFFF 300deg, #EEEFF4 300deg, #EEEFF4 360deg);
  border-radius: 40px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
}
.mainContainer .closeIcon {
  position: absolute;
  top: 24px;
  right: 24px;
  box-sizing: border-box;
  width: 24px;
  height: 24px;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}
.mainContainer .closeIcon:hover {
  background-color: #4BCEB5;
}
.mainContainer .mainContent {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: none;
}
.mainContainer .mainContent .textContent {
  text-align: center;
  margin: 31px 0 17px;
}
.mainContainer .mainContent .textContent h2 {
  font-family: "Helvetica Neue";
  font-style: normal;
  font-weight: 700;
  font-size: 36px;
  line-height: 38px;
  color: #123761;
  margin: 0 0 8px;
}
.mainContainer .mainContent .textContent p {
  font-family: "Helvetica Neue";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 17px;
  color: #848A9D;
  margin: 0;
}
.mainContainer .mainContent .buttonContainer {
  width: 100%;
}/*# sourceMappingURL=ModalMessageComponent.module.css.map */