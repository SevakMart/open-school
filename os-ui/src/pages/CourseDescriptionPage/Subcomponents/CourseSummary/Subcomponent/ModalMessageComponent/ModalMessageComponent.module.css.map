{"version":3,"sources":["ModalMessageComponent.module.scss","../../../../../../styles/Mixins/FlexMixins.scss","ModalMessageComponent.module.css","../../../../../../styles/Configs/Colors.scss","../../../../../../styles/Mixins/FontMixin.scss"],"names":[],"mappings":"AAMA;EACI,WAAA;EACA,YAAA;EACA,wIAAA;EACA,mBAAA;ECTF,aAAA;EACA,sBDSqB;ECRrB,mBDQ4B;ECP5B,6BDOmC;AEFrC;AFGC;EACC,kBAAA;EACA,SAAA;EACA,WAAA;EACA,sBAAA;EACA,WAAA;EACA,YAAA;ECjBA,aAAA;EACA,mBDiBmB;EChBnB,mBDgBuB;ECfvB,uBDe8B;EAC9B,eAAA;AEEF;AFDE;EACC,yBGrBe;ADwBlB;AFAI;ECxBF,aAAA;EACA,sBDwByB;ECvBzB,mBDuBgC;ECtBhC,qBDsBuC;AEKzC;AFJQ;EACI,kBAAA;EACA,mBAAA;AEMZ;AFLY;EI7BV,6BAAA;EACA,kBJ6BiD;EI5BjD,gBJ4BwD;EI3BxD,eJ2B4D;EI1B5D,iBJ0BiE;EACnD,cAAA;EACA,eAAA;AEWhB;AFTY;EIlCV,6BAAA;EACA,kBJkCiD;EIjCjD,gBJiCwD;EIhCxD,eJgC4D;EI/B5D,iBJ+BiE;EACnD,cAAA;EACA,SAAA;AEehB;AFZQ;EACI,WAAA;AEcZ","file":"ModalMessageComponent.module.css"}