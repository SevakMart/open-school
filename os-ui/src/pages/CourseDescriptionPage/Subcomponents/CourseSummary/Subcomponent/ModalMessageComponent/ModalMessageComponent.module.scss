@import "../../../../../../styles/Mixins/ButtonMixin.scss";
@import "../../../../../../styles/Mixins/FlexMixins.scss";
@import "../../../../../../styles/Mixins/FontMixin.scss";
@import "../../../../../../styles/Configs/Colors.scss";
@import "../../../../../../styles/Configs/Fonts.scss";

.mainContainer {
    width:100%;
    height:100%;
    background-image: conic-gradient(at 50% 38%,#EEEFF4 0deg,#EEEFF4 60deg,#FFFFFF 60deg,#FFFFFF 300deg,#EEEFF4 300deg,#EEEFF4 360deg);
    border-radius: 40px;
    @include flexStyle(column,center,space-evenly);
    .mainContent {
        @include flexStyle(column,center,none);
        .textContent {
            text-align: center;
            margin:31px 0 17px;
            h2 {
                @include fontStyle($secondary-font,normal,700,36px,38px);
                color:#123761;
                margin:0 0 8px;
            }
            p{
                @include fontStyle($secondary-font,normal,400,14px,17px);
                color:#848A9D;
                margin:0;
            }
        }
        .buttonContainer {
            width:100%;
        }
    }
}