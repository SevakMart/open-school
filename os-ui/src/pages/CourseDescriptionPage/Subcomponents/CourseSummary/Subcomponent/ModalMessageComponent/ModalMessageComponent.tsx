import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { closeModal } from '../../../../../../redux/Slices/PortalOpenStatus';
import EnrollIcon from '../../../../../../assets/svg/Enroll.svg';
import Button from '../../../../../../component/Button/Button';
import styles from './ModalMessageComponent.module.scss';
import CloseIcon from '../../../../../../icons/Close';

type Props = {
	courseId: number;
  };

const EnrolledSuccessMessage = ({ courseId }: Props) => {
  const {
    mainContainer, mainContent, textContent, buttonContainer, closeIcon,
  } = styles;
  const { t } = useTranslation();
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const handleClosePortal = () => {
    dispatch(closeModal());
  };

  const handleStartEnrolledCourse = () => {
    dispatch(closeModal());
    navigate(`/userCourse/modulOverview/${courseId}`);
  };

  return (
    <div className={mainContainer}>
      <div className={mainContent}>
        <div className={closeIcon} onClick={handleClosePortal}><CloseIcon /></div>
        <img src={EnrollIcon} alt="Enrolled" />
        <div className={textContent}>
          <h2>
            {t('string.courseDescriptionPage.title.modalSuccessTitle')}
            !
          </h2>
          <p>
            {t('string.courseDescriptionPage.text.modalSuccessText')}
            !
          </p>
        </div>
        <div className={buttonContainer}>
          <Button.MainButton className={['startEnrolledCourse']} onClick={handleStartEnrolledCourse}>
            {t('button.courseDescriptionPage.modalStartCourse')}
          </Button.MainButton>
        </div>
      </div>
    </div>
  );
};
export default EnrolledSuccessMessage;
