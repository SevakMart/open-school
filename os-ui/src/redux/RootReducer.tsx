import { combineReducers } from 'redux';
import chooseSubcategoryReducer from './Slices/ChoosSubcategorySlice';
import userInfoReducer from './Slices/loginUserSlice';
import allLearningPathFilterParamsReducer from './Slices/AllLearningPathFilterParamsSlice';
import forgotPasswordEmailReducer from './Slices/ForgotPasswordEmailSlice';
import signedUpUserIdReducer from './Slices/SignedUpUserIdSlice';
import suggestedCourseReducer from './Slices/SuggestedCourseSlice';
import userEnrolledCourseByCourseStatusReducer from './Slices/UserEnrolledCourseByCourseStatus';
import allLearningPathCoursesReducer from './Slices/AllLearningPathCourseSlice';
import allMentorsFilterParamsReducer from './Slices/AllMentorsFilterParamsSlice';
import allMentorsSliceReducer from './Slices/AllMentorsSlice';
import savedMentorsReducer from './Slices/SavedMentorsSlice';
import savedCourseReducer from './Slices/SavedLearningPathSlice';
import homepageMentorListReducer from './Slices/HomepageMentorSlice';
import portalStatusReducer from './Slices/PortalOpenStatus';

const rootReducer = combineReducers({
  portalStatus: portalStatusReducer,
  userInfo: userInfoReducer,
  chooseSubcategories: chooseSubcategoryReducer,
  filterParams: allLearningPathFilterParamsReducer,
  forgotPasswordEmail: forgotPasswordEmailReducer,
  signedUpUserId: signedUpUserIdReducer,
  suggestedCourse: suggestedCourseReducer,
  userEnrolledCourseByCourseStatus: userEnrolledCourseByCourseStatusReducer,
  allLearningPathCourses: allLearningPathCoursesReducer,
  allMentorsFilterParams: allMentorsFilterParamsReducer,
  allMentorsList: allMentorsSliceReducer,
  savedMentors: savedMentorsReducer,
  savedCourse: savedCourseReducer,
  homepageMentorList: homepageMentorListReducer,
});
export default rootReducer;
