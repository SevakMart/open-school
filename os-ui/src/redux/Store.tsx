import { configureStore } from '@reduxjs/toolkit';
import {
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import { createStateSyncMiddleware, initMessageListener } from 'redux-state-sync';
import rootReducer from './RootReducer';

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: {
      ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
    },
  }).concat(
    createStateSyncMiddleware({
      blacklist: ['persist/PERSIST', 'persist/REHYDRATE'],
    }),
  ),
});

initMessageListener(store);

export type RootState = ReturnType<typeof store.getState>
