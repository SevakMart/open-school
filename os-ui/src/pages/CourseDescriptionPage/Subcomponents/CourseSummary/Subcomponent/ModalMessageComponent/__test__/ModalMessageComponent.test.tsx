import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { store } from '../../../../../../../redux/Store';
import ModalMessageComponent from '../ModalMessageComponent';

describe('Create test cases for ModalMessageComponent', () => {
  test('Create a snapshot test', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ModalMessageComponent enrollInCourse={() => null} />
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
