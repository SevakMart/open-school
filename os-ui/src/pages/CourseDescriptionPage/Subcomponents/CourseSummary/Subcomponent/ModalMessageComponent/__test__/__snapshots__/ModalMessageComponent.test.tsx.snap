// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create test cases for ModalMessageComponent Create a snapshot test 1`] = `
<DocumentFragment>
  <div
    class="mainContainer"
  >
    <div
      class="mainContent"
    >
      <img
        alt="Enrolled"
        src="Enroll.svg"
      />
      <div
        class="textContent"
      >
        <h2>
          string.courseDescription.modal.congratsHeader
        </h2>
        <p>
          string.courseDescription.modal.mainMessage
        </p>
      </div>
      <button
        type="button"
      >
        button.startCourse
      </button>
    </div>
  </div>
</DocumentFragment>
`;
